tcms/rpc/tests/test_testcase.py

```diff
@@ -522,7 +606,26 @@ def test_add_component_from_another_product_is_not_allowed(self):
             self.rpc_client.TestCase.add_component(self.test_case.pk, self.bad_component.name)
 
 
-class TestCaseAddComment(APITestCase):
+class TestRemoveComponentPermission(APIPermissionsTestCase):
+    permission_label = 'testcases.delete_testcasecomponent'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+        self.test_case = TestCaseFactory()
+        self.good_component = ComponentFactory(product=self.test_case.category.product)
+        self.bad_component = ComponentFactory()
```

Comment:
`bad_component` is unused. You can assign it to the same TC and then assert that it is the only one left standing.